libsupport/include/katana/HostAllocator.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdlib>
+#include <cstring>
+
+#include "katana/config.h"
+
+namespace katana {
+
+enum class MemoryPinType { Swappable = 0, Pinned = 1 };
+
+class BaseHostAllocator {
+public:
+  virtual void* Malloc(const size_t n_bytes) const = 0;
+  virtual void* Calloc(const size_t n_items, const size_t item_size) const = 0;
+  virtual void* Realloc(void* ptr, const size_t n_bytes) const = 0;
+  virtual void Free(void* ptr) const = 0;
```

Comment:
Document these. It can mostly be just a reference to the standard library methods, but we should be explicit about that anyway. Since they have complex semantics (especially realloc).